### PR TITLE
Add FOSRestBundle recipe

### PR DESCRIPTION
**Pull request by 20uf**
| Q             | A
| ------------- | ---
| License       | MIT

About this bundle:
Is a popular bundle for the rapid development of RESTful applications and applications.
8.2 million downloads
https://github.com/FriendsOfSymfony/FOSRestBundle
@FriendsOfSymfony as the authors of the bundle

### CODE REVIEW

**Comment by GuilhemN** [comment]
I don't think we should include so many config options, imo only the most important should be there.

**Comment by 20uf** [comment]
@GuilhemN What determines the most important? Do you have an example?

**Comment by GuilhemN** [comment]
The one that are likely to be manually configured, so i would keep something minimal like:
```yml
fos_rest:
# Uncomment to activate features
#
#    param_fetcher_listener:  true
#    allowed_methods_listener:  true
#    routing_loader: true
#    view:
#        view_response_listener:  true
#    exception:
#        codes:
#            App\Exception\MyException: 403
#        messages:
#            App\Exception\MyException: Forbidden area.
#    format_listener:
#        rules:
#            - { path: ^/api, prefer_extension: true, fallback_format: json, priorities: [ json, html ] }
```

**Comment by lsmith77** [comment]
puh .. hard to say .. I tend to use quite a few settings in my projects. certainly I always use `body_listener` as well. but I would also say .. better to start small.

**Comment by GuilhemN** [comment]
Well indeed, I think we should add `body_listener`, removed by mistake :)

**Comment by 20uf** [comment]
@GuilhemN  @lsmith77 is it ok ?

File: friendsofsymfony/rest-bundle/2.2/etc/packages/fos_rest.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+#        view_response_listener:  true
+#    exception:
+#        codes:
+#            App\Exception\MyException: 403
+#        messages:
+#            App\Exception\MyException: Forbidden area.
+#    format_listener:
+#        rules:
+#            - { path: ^/api, prefer_extension: true, fallback_format: json, priorities: [ json, html ] }
+#    body_listener:
```
Comment (GuilhemN) [inline friendsofsymfony/rest-bundle/2.2/etc/packages/fos_rest.yaml]: Sub options should be removed imo.

**Comment by GuilhemN** [comment]
I wonder if we shouldn't uncomment everything, we would lose a bit in perf (we can probably improve that in the bundle) but dx would be greatly improved.

File: friendsofsymfony/rest-bundle/2.2/etc/packages/fos_rest.yaml (hunk 1 above)
Comment (20uf) [inline friendsofsymfony/rest-bundle/2.2/etc/packages/fos_rest.yaml]: just : 

```
#    body_listener:
#        default_format:       null
#        throw_exception_on_unsupported_content_type:  false`
```
?

**Comment by 20uf** [comment]
Tell me what you decide :)

File: friendsofsymfony/rest-bundle/2.2/etc/packages/fos_rest.yaml (hunk 1 above)
Comment (GuilhemN) [inline friendsofsymfony/rest-bundle/2.2/etc/packages/fos_rest.yaml]: I don't think these are very usual so I would just keep `true`.

**Comment by GuilhemN** [comment]
For me we should uncomment, at worse we'll still be able to revert if i'm the only thinking this way ☺

File: friendsofsymfony/rest-bundle/2.2/etc/packages/fos_rest.yaml (hunk 1 above)
Comment (20uf) [inline friendsofsymfony/rest-bundle/2.2/etc/packages/fos_rest.yaml]: Done

**Comment by GuilhemN** [comment]
Can you also do https://github.com/symfony/recipes-contrib/pull/47#issuecomment-306040966 ? :)

**Comment by 20uf** [comment]
@GuilhemN  What is the request your link does not work :(

**Comment by GuilhemN** [comment]
It was a link to my previous message:
> For me we should uncomment, at worse we'll still be able to revert if i'm the only thinking this way ☺

I think we should uncomment this default config.

File: friendsofsymfony/rest-bundle/2.2/etc/packages/fos_rest.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# Read the documentation: https://symfony.com/doc/master/bundles/FOSRestBundle/index.html
+fos_rest: ~
+# Uncomment to activate features
+#
+#    param_fetcher_listener:  true
+#    allowed_methods_listener:  true
+#    routing_loader: true
```
Comment (GuilhemN) [inline friendsofsymfony/rest-bundle/2.2/etc/packages/fos_rest.yaml]: My mistake, this doesn't exist... should be removed.

File: friendsofsymfony/rest-bundle/2.2/etc/packages/fos_rest.yaml (hunk 1 above)
Comment (GuilhemN) [inline friendsofsymfony/rest-bundle/2.2/etc/packages/fos_rest.yaml]: It seems it's enabled by default (https://github.com/FriendsOfSymfony/FOSRestBundle/blob/master/DependencyInjection/Configuration.php#L264) so it should be entirely removed... sorry one more change :stuck_out_tongue:

**Comment by 20uf** [comment]
I deleted the listener from the body. Le code is it appropriate?

```
# Read the documentation: https://symfony.com/doc/master/bundles/FOSRestBundle/index.html
fos_rest: ~
# Uncomment to activate features
#
#    param_fetcher_listener:  true
#    allowed_methods_listener:  true
#    routing_loader: true
#    view:
#        view_response_listener:  true
#    exception:
#        codes:
#            App\Exception\MyException: 403
#        messages:
#            App\Exception\MyException: Forbidden area.
#    format_listener:
#        rules:
#            - { path: ^/api, prefer_extension: true, fallback_format: json, priorities: [ json, html ] }

```

**Comment by 20uf** [comment]
@GuilhemN @lsmith77 It's OK for you ? If you agree, you can approve this PR?

**Comment by GuilhemN** [comment]
@20uf https://github.com/symfony/recipes-contrib/pull/47/files#r120013178 still need to be adressed and the file uncommented :)

**Comment by 20uf** [comment]
@GuilhemN It is done ! Tell me exactly what you want as configuration if it does not suit you, it will be simpler.

File: friendsofsymfony/rest-bundle/2.2/etc/packages/fos_rest.yaml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+# Read the documentation: https://symfony.com/doc/master/bundles/FOSRestBundle/index.html
+fos_rest: ~
+# Uncomment to activate features
```
Comment (fabpot) [inline friendsofsymfony/rest-bundle/2.2/etc/packages/fos_rest.yaml]: this can be removed

Comment (20uf) [inline friendsofsymfony/rest-bundle/2.2/etc/packages/fos_rest.yaml]: @fabpot want to tell you everything that is in comment is not worth it to be present?

Comment (fabpot) [inline friendsofsymfony/rest-bundle/2.2/etc/packages/fos_rest.yaml]: only this line which does not add any value IMHO (and the extra blank line as well)

Comment (20uf) [inline friendsofsymfony/rest-bundle/2.2/etc/packages/fos_rest.yaml]: It's done.

**Pull request merged**